lib/core/src/sdk.rs

```diff
@@ -2706,11 +2719,55 @@ impl LiquidSdk {
         Ok(config)
     }
 
+    async fn bip353_parse(&self, input: &str) -> Option<String> {
+        if let Some((local_part, domain)) = input.split_once('@') {
+            let resolver = Arc::clone(&self.dns_resolver);
+
+            let dns_name = format!("{}.{}.{}", local_part, USER_BITCOIN_PAYMENT_PREFIX, domain);
+
+            // Query for TXT records of a domain
+            let txt_data = match resolver.txt_lookup(dns_name).await {
+                Ok(records) => records
+                    .iter()
+                    .flat_map(|record| record.to_string().into_bytes())
+                    .collect::<Vec<u8>>(),
+                Err(e) => {
+                    eprintln!("Failed to lookup TXT records: {}", e);
+                    return None;
+                }
+            };
+
+            // Decode TXT data
+            match String::from_utf8(txt_data) {
+                Ok(decoded) => {
+                    if let Some((_, bolt12_address)) = decoded.split_once(BOLT12_PREFIX) {
+                        return Some(bolt12_address.to_string());
+                    }
+
+                    if let Some((_, lnurl)) = decoded.split_once(LNURL_PAY_PREFIX) {
+                        return Some(lnurl.to_string());
+                    }
+                }
+                Err(e) => {
+                    eprintln!("Failed to decode TXT data: {}", e);
+                }
```

Comment:
There is already some BIP 21 parsing logic in the core parser implementation. See [here](https://github.com/breez/breez-sdk-greenlight/blob/2208dd5530908d5cf7b607f34f80155669ab299b/libs/sdk-common/src/input_parser.rs#L202 ). It would be cool to offload BIP 21 parsing there.
At the moment, it only supports onchain addresses and bolt11 invoices. Bolt12 is currently only supported here in the liquid SDK, so it might still make sense to temporarilly have here a lookup for bolt12 offers.

Regarding LNURL, I'm not sure we need this. The point of BIP353 is to replace LNURL and not suffer from some of its drawbacks, so having an LNURL in a bip353 address would defeat the purpose.

Reply:
Ok I can move this inside the sdk_common so that it can be useful in both cases and next when also greenlight supports the bolt12 the feature is already implemented. I don't think that this could be a problem.

For the implementation of LNURL I followed the requests within the assignment of the planB tech school, I don't know why there is the need to support it in this form at the moment.

Reply:
Right, I was missing some context. Indeed we decided to look for LNURLs. Sorry for the confusion. In this case, it would be cool if LNURL parsing out of the BIP21 URI was in sdk-common such that it can also be used in the greenlight version.

Reply:
No problem. I am working in this direction, I will open PR on the greenlight version tomorrow I think